drake/solvers/Optimization.h

```diff
@@ -494,6 +541,13 @@ class DRAKEOPTIMIZATION_EXPORT OptimizationProblem {
       linear_constraints;  // note: does not include linear_equality_constraints
   std::list<Binding<LinearEqualityConstraint>> linear_equality_constraints;
   std::list<Binding<BoundingBoxConstraint>> bbox_constraints;
+
+  // An LCP encapsulates all of the constraints and objectives on a set of
+  // variables; if this is set, all of the above must be empty.
+  // TODO ggould this should be a boost::optional<> once boost is permitted.
+  std::unique_ptr<Binding<LinearComplementarityConstraint>>
+      linear_complementarity_constraint;
```

Comment:
i think that this should be a list of constraints, as above.  we will add many to a program (which can be assembled into one for the solve)